CodeJam.Main/Strings/StringBuilderExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Text;
+using System.Collections.Generic;
+
+using JetBrains.Annotations;
+
+namespace CodeJam.Strings
+{
+	/// <summary>
+	/// <see cref="StringBuilder"/> class extensions.
+	/// </summary>
+	[PublicAPI]
+	public class StringBuilderExtensions
+	{
+		/// <summary>
+		/// Determines whether the StringBuilder is empty.
+		/// </summary>
+		/// <param name="value">The StringBuilder to test.</param>
+		/// <returns>
+		/// <c>true</c> if <paramref name="value"/> is empty; otherwise, <c>false</c>.
+		/// </returns>
+		[Pure]
+		public static bool IsEmpty([NotNull] this StringBuilder value)
```

Comment:
I'd prefer to have a IsNullOrEmpty() method to keep symmetry with string and collection helpers.
And NotNullNorEmpty(), of course

Reply:
But StringBuilder instance cannot be null :)

Do you suggest ?
```cs
public static bool IsNullOrEmpty([CanBeNull] this StringBuilder value) 
```

Reply:
I don't see why we need the symmetry

Reply:
Another option is to use `Any()` similar to `IEnumerable.Any()` WDYT ?

Reply:
StringBuilder is a class and of course it can be null. So, `IsNullOrEmpty()` looks like a good candidate.

Reply:
Yes, but there is a null check with exception.
Ok. I can change to handle null too.

Reply:
I just don't see any usage for null.
Nobody writes methods with StringBuilder input, it will be string.
That's why string.IsNullOrEmpty is useful

Reply:
Well, look: the extension has to be useful. We do not add both IsEmpty() and IsNullOrEmpy() for string as it will confuse developers and may end with a mix of both methods.

Also, it is normal to use null references if the builder is cached somewhere. It allows to reduce amount of allocations and reduces memory pressure (a new builder allocates/borrows a buffer from the pool). So there may be null values and and it will not be an exceptional situation.